hathor/conf/settings.py

```diff
@@ -14,17 +14,20 @@
 
 import os
 from math import log
-from typing import List, NamedTuple, Optional
+from typing import Dict, List, Optional, Union
+
+from pydantic import validator
 
 from hathor.checkpoint import Checkpoint
+from hathor.utils import pydantic
 
 DECIMAL_PLACES = 2
 
 GENESIS_TOKEN_UNITS = 1 * (10**9)  # 1B
 GENESIS_TOKENS = GENESIS_TOKEN_UNITS * (10**DECIMAL_PLACES)  # 100B
 
 
-class HathorSettings(NamedTuple):
+class HathorSettings(pydantic.BaseModel):
```

Comment:
Is it immutable? Should we add a test to make sure?

Reply:
Changed as explained in https://github.com/HathorNetwork/hathor-core/pull/593#issuecomment-1546102674

Reply:
Great! Thanks!